File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:

        vowel_list=[]
        vowels="aeiouAEIOU"

        for c in s:
            if c in vowels:
                vowel_list.append(c)

        vowel_list.sort()
        result=""
        i=0
        for c in s:
            if c in vowels:
                result+=vowel_list[i]
                i+=1
            else:
                result+=c
        return result

                   

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        PriorityQueue<Character> pq=new PriorityQueue<>();

        for(char c:s.toCharArray()){
            if("aieouAEIOU".indexOf(c)!=-1){
                pq.offer(c);
            }
        }

        StringBuilder sb=new StringBuilder(s);

        for(int i=0;i<sb.length();i++){
            if("aeiouAEIOU".indexOf(sb.charAt(i))!=-1){
                sb.setCharAt(i,pq.poll());
            }
        }
        return sb.toString();
    }
}